### PR TITLE
create models and build relationships between them

### PR DESCRIPTION
**Pull request by vjnhat**

### CODE REVIEW

File: app/models/user_subject.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class UserSubject < ActiveRecord::Base
+belongs_to :user
```
Comment (dieunb) [inline app/models/user_subject.rb]: indent

File: app/models/user_subject.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+belongs_to :subject
```
Comment (dieunb) [inline app/models/user_subject.rb]: indent

File: app/models/user_subject.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+class UserSubject < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :subject
+
+  before_create :log_action
+  before_update :create_msg
+
+  private
+  def log_action
+   Activity.create user_id: user_id, action: "Was added to", target_id: subject_id
```
Comment (vjnhat) [inline app/models/user_subject.rb]: intent

File: app/models/course_user.rb [hunk 4]
```diff
@@ -1,4 +1,22 @@
...
   belongs_to :course
+
+  after_create :create_user_subject_task
+  after_create :log_action
+
+  private
+  def create_user_subject_task
+    @course = Course.find course_id 
+    @course.subjects.each do |subject|
+      UserSubject.create user_id: self.user_id, subject_id: subject.id, course_id: course_id, status: "Learning"
```
Comment (dieunb) [inline app/models/course_user.rb]: remove `self`

File: app/models/course_user.rb [hunk 5]
```diff
@@ -1,4 +1,22 @@
...
+      subject.tasks.each do |task|
+        UserTask.create user_id: self.user_id, task_id: task.id, course_id: course_id, status: "Can do"
```
Comment (dieunb) [inline app/models/course_user.rb]: remove `self`

File: app/models/activity.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+  def find_name
+    case action
+    when "Joined"
+      @course = Course.find target_id
+      @course.name
+    when "Was added to"
+      @subject = Subject.find target_id
+      @subject.name
+    when "Completed"
```
Comment (l3trungkjen) [inline app/models/activity.rb]: nên viết thế này hơn
when "Was added to", "Completed"
    @subject = Subject.find target_id
    @subject.name

File: app/models/user_task.rb [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+  after_create :log_action
+  after_update :create_msg
+
+  private
+  def log_action
+    create_activity user_id, "Can do", task_id
+  end
+
+  def create_msg
+    CreateActivity::create_activity user_id, "Did", task_id
```
Comment (dieunb) [inline app/models/user_task.rb]: ???

File: app/models/user_subject.rb [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+  after_create :log_action
+  after_update :create_msg
+
+  private
+  def log_action
+    create_activity user_id, "Was added to", subject_id
+  end
+
+  def create_msg
+    CreateActivity::create_activity user_id, "Completed", subject_id
```
Comment (dieunb) [inline app/models/user_subject.rb]: ???

File: app/models/activity.rb [hunk 9]
```diff
@@ -0,0 +1,23 @@
+class Activity < ActiveRecord::Base
+  belongs_to :user
+
+  def find_name
+    case action
```
Comment (dieunb) [inline app/models/activity.rb]: refactor again

File: app/models/user_subject.rb [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+
+  private
+  def log_action
+    create_activity user_id, "Was added to", subject_id
+  end
+
+  def create_msg
+    create_activity user_id, "Completed", subject_id
+  end
+
```
Comment (dieunb) [inline app/models/user_subject.rb]: remove this line

**Pull request merged**